dependencies.yaml

```diff
@@ -251,7 +251,7 @@ dependencies:
         packages:
           # Hard pin the patch version used during the build. This must be kept
           # in sync with the version pinned in get_arrow.cmake.
-          - pyarrow==12.0.1.*
+          - pyarrow==13.0.0.*
```

Comment:
Merge this with the `conda` dependency list above. Look at the previous PR to see how this was split up, and merge it back together by reverting those changes to the structure.